compiler/optimizer/OMRValuePropagation.cpp

```diff
@@ -1689,6 +1689,56 @@ TR_YesNoMaybe OMR::ValuePropagation::isCastClassObject(TR::VPClassType *type)
    return TR_maybe;
    }
 
+
+TR_YesNoMaybe OMR::ValuePropagation::isArrayCompTypeValueType(TR::VPConstraint *arrayConstraint)
+   {
+   TR_YesNoMaybe isValueType = TR_maybe;
+
+   if (!TR::Compiler->om.areValueTypesEnabled())
+      {
+      isValueType = TR_no;
+      }
+   else if (!(arrayConstraint && arrayConstraint->getClass()
+              && arrayConstraint->getClassType()->isArray() == TR_yes))
+      {
+      isValueType = TR_maybe;
```

Comment:
Because `isValueType` is initialized as `TR_maybe` already, maybe only the cases where it has be set as `TR_yes` or `TR_no` are required. Perhaps most of the places where  `TR_maybe` is explicitly set could be eliminated. Just need to be careful of the `if` and `else` condition checks.

Reply:
Thinking about this suggestion a little further, I would either leave the conditions in place and have empty blocks in place of the current locations that set `isValueType = TR_maybe`, which is a little bit ugly, or the conditions would change to look like this, which I think makes it slightly more difficult to think about the `TR_maybe` cases:

```
else if (arrayConstraint && arrayConstraint->getClass()
              && arrayConstraint->getClassType()->isArray() == TR_yes)
   {
   TR_OpaqueClassBlock *arrayComponentClass = fe()->getComponentClassFromArrayClass(arrayConstraint->getClass());
   if (arrayComponentClass)
      {
      if (TR::Compiler->cls.isValueTypeClass(arrayComponentClass))
         {
         isValueType = TR_yes;
         }
      else if (TR::Compiler->cls.isClassArray(comp(), arrayComponentClass))
         {
         isValueType = TR_no;
         }
      else if (TR::Compiler->cls.isConcreteClass(comp(), arrayComponentClass))
         {
         int32_t len;
         const char *sig = arrayConstraint->getClassSignature(len);

         if (sig && arrayConstraint->isFixedClass() && sig[0] == '[' && len == 19
             && !strncmp(sig, "[Ljava/lang/Object;", 19))
            {
            isValueType = TR_no;
            }
         }
      }
   }
```

Maybe it would be better if I dropped the explicit initialization of `isValueType` on its declaration, or replaced the use of `isValueType` with `return` statements.  Any thoughts on these choices, Annabelle @a7ehuo?